docs/tsp/typeServerProtocol.ts

```diff
@@ -230,6 +244,20 @@ export namespace TypeServerProtocol {
         snapshot: number;
     }
 
+    export interface GetTypeAttributeParams {
+        // The type for which the attribute is being requested.
+        type: Type;
+        // The name of the attribute being requested.
+        attributeName: string;
+        // Flags that control how the attribute is accessed.
+        accessFlags: AttributeAccessFlags;
+        // Optional: The expression node that the member is being accessed from.
+        expressionNode?: Node;
```

Comment:
when is this necessary? why is it necessary?

Reply:
It gives a clue to the type checker on how to look up the attribute. Specifically, when looking up bound members. 

For example, this class heirarchy:

```python
class A:
    def foo(self):
        pass
        
class B(A):
     def foo(self):
         pass
         
b: A = B()
b.foo() # Get type attributes for foo here
```

Reply:
in this case, we can figure this out with just type: `B` and attributeName: `foo`. what extra information does expression node provide?

Reply:
Maybe that was a bad example, this is more of the case that hits:

```python
class A:
    def __bool__(self) -> bool:
        return True
        
a = bool(A()) # Goto def on bool here
print(a)
```

Internally we search for the `__bool__` method. It needs the A() in the expression to know what to search for.

Reply:
The `bool(A())` is the expression node in this case.